website/source/guides/operations/vault-ha-consul.html.md

```diff
@@ -0,0 +1,621 @@
+implementation. While this is not an exhaustive or prescriptive guide that can
+be used as a drop-in production example, it covers the basics enough to inform
+your own production setup.
+
+
+## Reference Materials
+
+- [High Availability Mode](/docs/concepts/ha.html)
+- [Consul Storage Backend](/docs/configuration/storage/consul.html)
+- [High Availability Parameters](/docs/configuration/index.html#high-availability-parameters)
+- [Consul Agent Configuration](https://www.consul.io/docs/agent/options.html)
+
+## Estimated Time to Complete
+
+10 minutes
+
+## Prerequisites
+
+- Vault
+- Consul
+
+## Steps
+
+Our goal in following this guide is to arrive at a Vault HA setup
+consisting of the following:
+
+- 2 Vault servers: 1 active and 1 standby
+- Cluster of 3 Consul servers
+
+### Reference Diagram
+
+This diagram lays out the simple architecture details for reference:
+
+![Reference Architecture](/assets/images/vault-ha-consul.png)
+
+We’ll use the open source software editions of Vault and Consul in this guide,
+but the setup is the same for Enterprise editions.
+
+### Consul Server Cluster
+
+Our Consul servers in this guide will be defined by IP address only, but also
+referenced by a label:
+
+- **`consul_s1: 10.1.42.101`**
+- **`consul_s2: 10.1.42.102`**
+- **`consul_s3: 10.1.42.103`**
+
+The consul binary is presumed to be located in /usr/local/bin/consul, but if
+your case differs, you can adjust the path references accordingly.
+
+With that in mind, here is a basic Consul server configuration starting point:
+
+#### Consul Server Configuration
+
+```plaintext
+{
+  "server": true,
+  "node_id": "$NODE_ID",
+  "datacenter": "dc1",
+  "data_dir": "$CONSUL_DATA_PATH",
+  "bind_addr": "0.0.0.0",
+  "client_addr": "0.0.0.0",
+  "advertise_addr": "$ADVERTISE_ADDR",
+  "bootstrap_expect": 3,
+  "retry_join": ["$JOIN1", "$JOIN2", "$JOIN3"],
+  "ui": true,
+  "log_level": "DEBUG",
+  "enable_syslog": true,
+  "acl_enforce_version_8": false
+}
+```
+
+Note that some values contain variable placeholders while the rest have
+reasonable defaults. You should change the following in your own Consul server
+configuration based on the example:
+
+- `$CONSUL_DATA_PATH`: absolute path to Consul data directory; ensure that
+this directory is writable by the Consul process user.
+- `$ADVERTISE_ADDR`: this should be set to address that you prefer the Consul
+servers advertise to the other servers in the cluster and should not be set to
+`0.0.0.0`; for this guide, it should be set to the Consul server’s IP address in
+each instance of the configuration file, or `10.1.42.101`,` 10.1.42.102`, and
+`10.1.42.103` respectively.
+- `JOIN1**`, `**JOIN2`, `$JOIN3`: This example uses the `retry_join`
+method of joining the server agents to form a cluster; as such, the values for
+this guide would be `10.1.42.101`, `10.1.42.102`, and `10.1.42.103` respectively.
+- `$NODE_ID` this is a unique label for the node; in our case, this will be
```

Comment:
This just needs updating to `$NODE_NAME` instead; the descriptive text should be good still.